src/components/AttachmentPicker/index.native.js

```diff
@@ -94,43 +119,117 @@ function getDataForUpload(fileData) {
     };
 }
 
-const propTypes = {
-    children: PropTypes.func.isRequired,
-};
-
 /**
  * This component renders a function as a child and
  * returns a "show attachment picker" method that takes
- * a callback. This is the web/mWeb/desktop version since
- * on iOS Safari we must append a hidden input to the DOM
- * and listen to onChange event. When the show method is
- * called an attachment
- *
- * @example
- * <AttachmentPicker>
- * {({openPicker}) => (
- *     <Button
- *         onPress={() => {
- *             openPicker({
- *                 onPicked: (file) => {
- *                     // Display or upload File
- *                 },
- *             });
- *         }}
- *     />
- * )}
- * </AttachmentPicker>
- *
- * @returns {Function}
+ * a callback. This is the ios/android implementation
+ * opening a modal with attachment options
  */
-const AttachmentPicker = ({children}) => children({
-    openPicker: ({onPicked}) => {
-        show((response) => {
-            onPicked(getDataForUpload(response));
+class AttachmentPicker extends Component {
+    constructor(...args) {
+        super(...args);
+
+        this.state = {isVisible: false, onPicked: () => {}, result: null};
+
+        this.menuItemData = [
+            {
+                icon: Camera,
+                text: 'Take Photo',
+                pickAttachment: () => showImagePicker(RNImagePicker.launchCamera),
+            },
+            {
+                icon: Gallery,
+                text: 'Choose from Gallery',
+                pickAttachment: () => showImagePicker(RNImagePicker.launchImageLibrary),
+            },
+            {
+                icon: Paperclip,
+                text: 'Choose Document',
+                pickAttachment: showDocumentPicker,
+            },
+        ];
+
+        this.setResult = this.setResult.bind(this);
+        this.close = this.close.bind(this);
+        this.onModalHide = this.onModalHide.bind(this);
+    }
+
+    /**
+     * After the modal closes, if an attachment was selected delegate it to the `onPicked` callback
+     */
+    onModalHide() {
```

Comment:
Functions should be named for what it does, not what triggered it. Something like `setAttachment`?

Reply:
or `setResult`

Reply:
> It's not something that you can call anytime, it needs to be called specifically when the modal is fully gone 

I think it's fine to note that in the function comment. I'm not sure that the CreateMenu usage is valid tbh. It's also the same name as a ReactNativeModal prop, which is a little confusing.

`completeAttachmentSelection` seems fine

Reply:
Renamed to `completeAttachmentSelection` leaving unresolved so you can verify this fix